src/front/wgsl/mod.rs

```diff
@@ -3295,13 +3305,69 @@ impl Parser {
                     crate::Expression::Load {
                         pointer: reference.handle,
                     },
-                    NagaSpan::from(span_start..lexer.current_byte_offset()),
+                    lhs_span.into(),
                 );
                 let right = self.parse_general_expression(lexer, context.reborrow())?;
                 context
                     .expressions
                     .append(crate::Expression::Binary { op, left, right }, span.into())
             }
+            token @ (Token::IncrementOperation, _) | token @ (Token::DecrementOperation, _) => {
+                let op = match token.0 {
+                    Token::IncrementOperation => Bo::Add,
+                    Token::DecrementOperation => Bo::Subtract,
+                    _ => unreachable!(),
+                };
+                let op_span = token.1;
+
+                // prepare the typifier, but work around mutable borrowing...
+                let _ = context.resolve_type(reference.handle)?;
+
+                let ty = context.typifier.get(reference.handle, context.types);
+                let constant_inner = match ty.canonical_form(context.types) {
+                    Some(crate::TypeInner::ValuePointer {
+                        size: None,
+                        kind,
+                        width,
+                        space: _,
+                    }) => crate::ConstantInner::Scalar {
+                        width,
+                        value: match kind {
+                            crate::ScalarKind::Sint => crate::ScalarValue::Sint(1),
+                            crate::ScalarKind::Uint => crate::ScalarValue::Uint(1),
+                            _ => {
+                                return Err(Error::BadIncrDecrReferenceType(lhs_span));
+                            }
+                        },
+                    },
+                    _ => {
+                        return Err(Error::BadIncrDecrReferenceType(lhs_span));
+                    }
+                };
+                let constant = context.constants.append(
+                    crate::Constant {
+                        name: None,
+                        specialization: None,
+                        inner: constant_inner,
+                    },
+                    crate::Span::default(),
+                );
+
+                let left = context.expressions.append(
+                    crate::Expression::Load {
+                        pointer: reference.handle,
+                    },
+                    lhs_span.into(),
+                );
+                let right = context.interrupt_emitter(
```

Comment:
why are we interrupting the emitter here?

Reply:
I kept getting this validation error without it

```
thread 'convert_wgsl' panicked at 'Naga module validation failed: WithSpan { inner: EntryPoint { stage: Fragment, name: "fs_main", error: Function(ExpressionAlreadyInScope([22])) }, spans: [] }', tests/snapshots.rs:108:10
```

and I've seen it used here most likely for the same reason

https://github.com/gfx-rs/naga/blob/cf992a9a95026d8da2f4847e4884e6c0efbba1db/src/front/wgsl/mod.rs#L2301